src/coreclr/tests/src/tracing/eventpipe/gcdump/gcdump.cs

```diff
@@ -0,0 +1,107 @@
+using System.Diagnostics.Tracing;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Microsoft.Diagnostics.NETCore.Client;
+using Microsoft.Diagnostics.Tools.RuntimeClient;
+using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Parsers;
+using Tracing.Tests.Common;
+using Microsoft.Diagnostics.Tracing.Parsers.Clr;
+
+namespace Tracing.Tests.EventSourceError
+{
+    public class GCDumpTest
+    {
+        private static int _bulkTypeCount = 0;
+        private static int _bulkNodeCount = 0;
+        private static int _bulkEdgeCount = 0;
+        private static int _bulkRootEdgeCount = 0;
+        private static int _bulkRootStaticVarCount = 0;
+
+        private static readonly ulong GC_HeapDump_Keyword = 0x100000UL;
+
+        public static int Main(string[] args)
+        {
+            // This test validates that if an EventSource generates an error
+            // during construction it gets emitted over EventPipe
+
+            List<Provider> providers = new List<Provider>
+            {
+                new Provider("Microsoft-Windows-DotNETRuntime", eventLevel: EventLevel.Verbose, keywords: (ulong)ClrTraceEventParser.Keywords.GCHeapSnapshot)
+            };
+
+            var configuration = new SessionConfiguration(circularBufferSizeMB: 1024, format: EventPipeSerializationFormat.NetTrace,  providers: providers);
+            return IpcTraceTest.RunAndValidateEventCounts(_expectedEventCounts, _eventGeneratingAction, configuration, _DoesRundownContainMethodEvents);
+        }
+
+        private static Dictionary<string, ExpectedEventCount> _expectedEventCounts = new Dictionary<string, ExpectedEventCount>()
+        {
+            // This space intentionally left blank
+        };
+
+        private static Action _eventGeneratingAction = () =>
+        {
+            // This space intentionally left blank
+        };
+
+        private static Func<EventPipeEventSource, Func<int>> _DoesRundownContainMethodEvents = (source) =>
+        {
+            source.Clr.TypeBulkType += (GCBulkTypeTraceData data) =>
+            {
+                _bulkTypeCount += data.Count;
+            };
+
+            source.Clr.GCBulkNode += delegate (GCBulkNodeTraceData data)
+            {
+                _bulkNodeCount += data.Count;
+            };
+
+            source.Clr.GCBulkEdge += (GCBulkEdgeTraceData data) =>
+            {
+                _bulkEdgeCount += data.Count;
+            };
+
+            source.Clr.GCBulkRootEdge += (GCBulkRootEdgeTraceData data) =>
+            {
+                _bulkRootEdgeCount += data.Count;
+            };
+
+            source.Clr.GCBulkRootStaticVar += (GCBulkRootStaticVarTraceData data) =>
+            {
+                _bulkRootStaticVarCount += data.Count;
+            };
+
+            return () => 
+            {
+                // 50 is a safe number, comfortably less than the events we expect.
```

Comment:
Could you put in the comment how many we "expect"?  It will make diagnosing test failures easier later on.

Reply:
I was thinking when I chose 50 that I basically want to verify that there is more than 1, but I don't really want to be in the business of tracking how many there should be. I expect that number to change over time since most of the types will be in framework code. As people add and remove types the numbers could change a lot over time.

How are you envisioning knowing the amount of types would help with debugging?

Reply:
I was envisioning situations where we get "some" events but less than 50.  If we really only care that we get _any_ I was thinking we should have the test just check for `>1`, but I think this is probably fine as is.

Reply:
I would like to be as close to accurate as possible but also not have to deal with what happens when it changes a little bit. Maybe 50 is too low. I was originally thinking I wanted to avoid the scenario where there is some small number of hardcoded types and we miss the dynamic ones.

I was already waffling about whether to just go with 1 or try to make it close to the real values before your feedback.

These are the current values
```
_bulkTypeCount =          153
_bulkNodeCount =          724
_bulkEdgeCount =          1066
_bulkRootEdgeCount =      321
_bulkRootStaticVarCount = 92
```

Instead I'll change it to be ~80% of those numbers and add a comment that if it becomes and sort of maintenance burden just change the values to 1

Reply:
Nevermind, it already was causing failures so I switched it back